agent/consul/intention_endpoint.go

```diff
@@ -25,6 +25,17 @@ type Intention struct {
 	srv *Server
 }
 
+func (s *Intention) checkIntentionID(id string) (bool, error) {
+	state := s.srv.fsm.State()
+	if _, ixn, err := state.IntentionGet(nil, id); err != nil {
+		return false, err
+	} else if ixn != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
 // Apply creates or updates an intention in the data store.
```

Comment:
`Apply` is a pretty considerable fn here -- I would recommend expanding the doccomment to describe its expected usage, side effects, and gotchas/footguns that might catch future folks by surprise if there are any.

Reply:
Unfortunately the doc says exactly what it does. The entire body here is just how we go about doing it. I thought about refactoring this into a single switch statement on the Op field of the RPC request that called one of 3 other funcs but didn't do it because it would be a larger change and unnecessary for the work I am doing.

The logic in here is weird because ACL enforcement and request validation are done a bit differently depending on the Op. Maybe I will refactor into 3 funcs.

Reply:
Makes sense -- and I agree that a refactor here is pretty big and probably out of scope of this task. If you do find yourself coming back to this function, I'd be interested in pairing with you to learn from how you'd approach the refactor!

Reply:
@mkcp Let me know what you think of the new Intention.Apply RPC endpoint. I refactored things to do the Op switch and then offload to a op specific function. This removed much of the conditional branching in this RPC that I thought made it hard to follow.

It does duplicate a few things but IMO, I like the new way better. What do you think?

Reply:
I think the refactor works great! 💯  The control flow of the function is clear at a glance, and the names you chose help clearly delineate the branches.